src/styles/_thermostat.scss

```diff
@@ -1,6 +1,200 @@
 @use './colors';
 
-@mixin all {
+@mixin temperature-control-thumb($primary-color: colors.$thermo-blue) {
+  width: var(--thumb-size);
+  height: var(--thumb-size);
+  border-radius: 1em;
+  background: colors.$white;
+  border: 2px solid $primary-color;
+  box-shadow: none;
+}
+
+@mixin temperature-control-track($primary-color: colors.$thermo-blue) {
+  height: var(--track-height);
+  border-radius: 0.5em;
+  background:
+    linear-gradient($primary-color, $primary-color) 0 / var(--slider-position)
+      100% no-repeat,
+    colors.$text-gray-3;
+  border: none;
+  box-shadow: none;
+}
+
+@mixin temperature-control {
+  .seam-temperature-control {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    flex-direction: row;
+    gap: 7px;
+  }
+
+  .seam-temperature-range-wrap {
+    --temperature-min: 0;
+    --temperature-max: 100;
+    --temperature-current: 50;
+
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    position: relative;
+  }
+
+  input[type='range'].seam-temperature-range {
+    --temperature-range: calc(var(--temperature-max) - var(--temperature-min));
+    --temperature-ratio: calc(
+      (var(--temperature-current) - var(--temperature-min)) /
+        var(--temperature-range)
+    );
+    --slider-position: calc(
+      0.5 * 20px + var(--temperature-ratio) * (100% - 20px)
+    );
+
+    --thumb-size: 20px;
+    --track-height: 4px;
+
+    -webkit-appearance: none;
+
+    &:focus {
+      outline: none;
+    }
+
+    &[data-variant='heat'] {
+      // Thumbs
+      &::-webkit-slider-thumb {
+        @include temperature-control-thumb(colors.$thermo-orange);
+      }
+      &::-moz-range-thumb {
+        @include temperature-control-thumb(colors.$thermo-orange);
+      }
+      &::-ms-thumb {
+        @include temperature-control-thumb(colors.$thermo-orange);
+      }
+
+      // Tracks
+      &::-webkit-slider-runnable-track {
+        @include temperature-control-track(colors.$thermo-orange);
+      }
+      &::-moz-range-track {
+        @include temperature-control-track(colors.$thermo-orange);
+      }
+      &::-ms-track {
+        @include temperature-control-track(colors.$thermo-orange);
+      }
+
+      // etc.
+      &::-ms-fill-lower {
+        background: colors.$thermo-orange;
+      }
+    }
+
+    &[data-variant='cool'] {
+      // Thumbs
+      &::-webkit-slider-thumb {
```

Comment:
We can't combine the selectors here; i.e.

```scss
&::-webkit-slider-thumb,
&::-moz-range-thumb,
&::-ms-thumb {
  // ...
}
```

However, using a mixin is a fair compromise for reduction of repetition.